packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -7,6 +7,8 @@ import 'image_stream.dart';
 const int _kDefaultSize = 1000;
 const int _kDefaultSizeBytes = 100 << 20; // 100 MiB
 
+typedef LargeImageHandler = void Function(ImageCache, int);
```

Comment:
This needs a docstring.

I'm not sure if this is entirely clear on how it will be used. Do we normally expect this to mutate the cache parameters (e.g. raise the `maxSizeBytes`)?  Are there gotchas that we need to guard against here (e.g. someone calls `putIfAbsent` in this callback until a `StackOverflowError` occurs, or mutates state in some other unexpected way)?

I wonder if this could return a type instead of just mutating the cache.  Someone could get a hold of the cache in this method, but we could document it as saying you should avoid doing that and instead just fill out the information on the type (such as `newMaxSizeInBytes`).  Or at the very least, if there's just some way we can steer users towards the correct/performant behavior.

Reply:
> This needs a docstring.
> 
Docstrings aren't required on typedefs.  I think the questions you raise are better suited on the documentation for the field.

> I'm not sure if this is entirely clear on how it will be used. Do we normally expect this to mutate the cache parameters (e.g. raise the `maxSizeBytes`)?

I've reworded the docstring to address this.

> Are there gotchas that we need to guard against here (e.g. someone calls `putIfAbsent` in this callback until a `StackOverflowError` occurs, or mutates state in some other unexpected way)?

All callbacks have the potential for infinite recursion but we don't document them like that.

> I wonder if this could return a type instead of just mutating the cache. Someone could get a hold of the cache in this method, but we could document it as saying you should avoid doing that and instead just fill out the information on the type (such as `newMaxSizeInBytes`). Or at the very least, if there's just some way we can steer users towards the correct/performant behavior.

We could make the function return a new maximumByteSize.  I wonder if that nudges users to think they should resize it.

Reply:
If it's a public type it really should have a docstring - it doesn't have to be complicated (see e.g. https://api.flutter.dev/flutter/dart-ui/VoidCallback.html or https://api.flutter.dev/flutter/dart-ui/FrameCallback.html). 

I'm not sure returning an int would be enough if the user wants to resize and try with the resized image though.  I'm thinking something that gives them enough info to say how many bytes are left in the cache, which they could then use to calculate a reasonable size of the image to resize to - or maybe some combination of increasing cache size + downsizing image

Reply:
> If it's a public type it really should have a docstring

Done.

Reply:
> I'm not sure returning an int would be enough if the user wants to resize and try with the resized image though. I'm thinking something that gives them enough info to say how many bytes are left in the cache, which they could then use to calculate a reasonable size of the image to resize to - or maybe some combination of increasing cache size + downsizing image

That's not going to be possible to retry with another image the way this whole thing is plumbed.  The caching of images is hooked onto the successful loading of images, we can't stop or change the loading of the successful image.  I think it's a good idea but it is non-trivial and not one of the 2 use cases we had to accommodate.